docs/contributing/backend.md

```diff
@@ -0,0 +1,272 @@
+# Contributing to Nixopus Backend
+
+This guide provides detailed instructions for contributing to the Nixopus backend codebase.
+
+## Setup for Backend Development
+
+1. **Prerequisites**
+   - Go version 1.23.6 or newer
+   - PostgreSQL
+   - Docker and Docker Compose (recommended)
+
+2. **Environment Setup**
+
+   ```bash
+   # Clone the repository
+   git clone https://github.com/raghavyuva/nixopus.git
+   cd nixopus
+   
+   # Set up PostgreSQL database
+   createdb nixopus -U postgres
+   createdb nixopus_test -U postgres
+   
+   # Copy environment template
+   cp api/.env.sample api/.env
```

Comment:
Ensure we have a .env.sample inside api folder, or create one if necessary. we already have .env.sample which serves as a global environment sample file